src/ci/docker/run.sh

```diff
@@ -17,6 +17,8 @@ dist=$objdir/build/dist
 
 source "$ci_dir/shared.sh"
 
+CACHE_DOMAIN="${CACHE_DOMAIN:-ci-caches.rust-lang.org}"
```

Comment:
Can we default to the S3 bucket domain here? This way the change will work on the temporary GHA environment, which doesn't have any CDN for caches.

Reply:
GHA should have a CDN for caches? I believe ci-caches-gha.rust-lang.org is setup to point to the temporary S3 bucket for GHA eval...

Reply:
Oh I totally forgot we did set that up as well.